### PR TITLE
[YUNIKORN-2719]Insert invalid group name check

### PR DESCRIPTION
**Pull request by SophieTech88**
### What is this PR for?
Assert invalid group name in Get Group REST API

1.  Add the invalid group name check in handlers.go. 
2. Add assertInvalidGroupName func in handlers_test.go.
3. Add a new test case for invalid group name according to GroupRegExp
4. Fix the bug for assertGroupNotExists in one test case.

### What type of PR is it?
* [x] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2719

### How should this be tested?
It passed all unit tests locally.

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Comment by chia7712** [comment]
The failed test is related to https://github.com/apache/yunikorn-core/pull/905/files#r1678723246

**Comment by SophieTech88** [comment]
Finally find， this commit can't pass the unit test, because the `TestSpecificGroupResourceUsage` failed.  

And the error message is `YUNIKORN/yunikorn-core/pkg/webservice/handlers_test.go:1785: assertion failed: 404 (http.StatusNotFound int) != 400 (resp.statusCode int): Incorrect Status code`. 

Then dig into it, find in TestSpecificGroupResourceUsage, there is a test case as below:

``// Test group name with special characters escaped properly
	validGroup := url.QueryEscape("test_a-b_c@#d@do:mai/n.com")
	req, err = createRequest(t, "/ws/v1/partition/default/usage/group", map[string]string{"user": "testuser", "group": validGroup})
	assert.NilError(t, err)
	resp = &MockResponseWriter{}
	getGroupResourceUsage(resp, req)
	assertGroupNotExists(t, resp)
```
this group name `test_a-b_c@#d@do:mai/n.com` can't pass the InvalidGroupName test, so it returns 400 bad request error, which is different with the GroupNotExists error 404.

**Comment by chia7712** [comment]
> this group name test_a-b_c@#d@do:mai/n.com can't pass the InvalidGroupName test, so it returns 400 bad request error, which is different with the GroupNotExists error 404.

Yep, maybe we can assume `test_a-b_c@#d@do:mai/n.com` is a test case which should fail on RegExp. Hence, please try to update `assertGroupNotExists` to make test pass

**Comment by SophieTech88** [comment]
Many thanks for your suggestion. Update a new commit to fix the issue. Now it works as expected.

**Comment by chia7712** [comment]
@SophieTech88 please fix the conflicts!

File: pkg/webservice/handlers_test.go [hunk 1]
```diff
@@ -2007,6 +2016,14 @@ func TestSpecificGroupResourceUsage(t *testing.T) {
...
 	assert.Equal(t, errInfo.Message, "invalid URL escape \"%Zt\"", jsonMessageError)
 	assert.Equal(t, errInfo.StatusCode, http.StatusBadRequest)
+
+	// Test invalid group name according to GroupRegExp
+	invalidGroupName := "invalid_group@name"
+	req, err = createRequest(t, "/ws/v1/partition/default/usage/group", map[string]string{"user": "testuser", "group": invalidGroupName})
+	assert.NilError(t, err)
+	resp = &MockResponseWriter{}
+	getGroupResourceUsage(resp, req)
+	assertInvalidGroupName(t, resp)
```
Comment (baconYao) [inline pkg/webservice/handlers_test.go:2026]: I think this piece of code duplicates another [one](https://github.com/apache/yunikorn-core/pull/905/files#diff-c547735b58862d5573852f5e2c6a17c723f81a5620f3658172dd1b321f035975R2016-R2021).

I think the change of line [2001](https://github.com/apache/yunikorn-core/pull/917/files#diff-c547735b58862d5573852f5e2c6a17c723f81a5620f3658172dd1b321f035975R2001) is enough to cover what you implemented in [handlers.go](https://github.com/apache/yunikorn-core/pull/917/files#diff-c547735b58862d5573852f5e2c6a17c723f81a5620f3658172dd1b321f035975R2001)

**Review by baconYao — COMMENTED** [review]
Hi @SophieTech88, please see my comment. Thanks.

File: pkg/webservice/handlers_test.go (hunk 1 above)
Comment (SophieTech88) [reply pkg/webservice/handlers_test.go]: Many thanks for your suggestion. The test for InvalidGroupName is already exist. So I just deleted the new created one. Update the PR.

**Review by chenyulin0719 — APPROVED** [review]
+1 LGTM

**Review by pbacsko — APPROVED** [review]
LGTM

**Review by chia7712 — APPROVED** [review]

**Pull request closed without merging**